webapp/src/main/java/org/apache/atlas/web/rest/TypesREST.java

```diff
@@ -459,6 +461,25 @@ public AtlasTypesDef createAtlasTypeDefs(final AtlasTypesDef typesDef) throws At
         }
     }
 
+    private void validateTypeCreateOrUpdate(AtlasTypesDef typesDef) throws AtlasBaseException {
+
+        if (CollectionUtils.isNotEmpty(typesDef.getEnumDefs())) {
+            AtlasBaseTypeDef type = typesDef.getEnumDefs().stream().filter(typeDefStore::validateTypeName).findFirst().orElse(null);
+            if (type != null)
+                throw new AtlasBaseException(AtlasErrorCode.FORBIDDEN_TYPENAME, type.getName());
+        }
+        if (CollectionUtils.isNotEmpty(typesDef.getEntityDefs())) {
+            AtlasBaseTypeDef type = typesDef.getEntityDefs().stream().filter(typeDefStore::validateTypeName).findFirst().orElse(null);
```

Comment:
instead of .findFirst().orElse(null) use anyMatch would be minor change but checking null is looking antipattern.